File: core/src/main/java/com/serious/service/ComponentHost.java
package com.serious.service;
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import org.springframework.stereotype.Component;

import java.lang.annotation.*;

/**
 * @author Andreas Ernst
 */
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Component
public @interface ComponentHost {
    String[] channels() default {"local"};

    String health() default "/health";
}

File: core/src/main/java/com/serious/service/ComponentHost.kt
package com.serious.service

import org.springframework.stereotype.Component

/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

/**
 * Marks [Component] implementations
 */
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
@Component
annotation class ComponentHost(val health: String = "/health") 